src/js/rtp-stats.js

```diff
@@ -5,86 +5,123 @@
                         timestamp:          timestamp,
                         packetsCount:       statsReport.packetsSent,
                         bytesSent:          statsReport.bytesSent,
                         packetsLost:        is_defined(statsReport.packetsLost) ? Math.max(0, statsReport.packetsLost) : 0,
                         rttMilliseconds:    when_defined(statsReport.googRtt),
                         procMilliseconds:   is_defined(statsReport.googCurrentDelayMs),
                         frameRateSent:      when_defined(statsReport.googFrameRateSent),
-                    };
+                    }, statsReport.type, streamType);
 
                 } else if (typeof statsReport.packetsReceived !== 'undefined' && statsReport.mediaType == 'video' && streamType === 'video_output') {
-                    extractedStats = {
+                    extractedStats = new MediaRtpStats({
                         timestamp:          timestamp,
                         packetsCount:       statsReport.packetsSent,
                         bytesReceived:      statsReport.bytesReceived,
                         packetsLost:        is_defined(statsReport.packetsLost) ? Math.max(0, statsReport.packetsLost) : 0,
                         frameRateReceived:  when_defined(statsReport.googFrameRateReceived),
                         procMilliseconds:   is_defined(statsReport.googCurrentDelayMs),
                         jbMilliseconds:     when_defined(statsReport.googJitterBufferMs)
-                    };
+                    }, statsReport.type, streamType);
 
                 }
             } else if (statsReport.type === 'inboundrtp') {
                 // Firefox case. Firefox reports packetsLost parameter only in inboundrtp type, and doesn't report in outboundrtp type.
                 // So we only pull from inboundrtp. Firefox reports only stats for the stream passed in.
                 if (is_defined(statsReport.packetsLost) && is_defined(statsReport.packetsReceived)) {
-                    extractedStats = {
+                    extractedStats = new MediaRtpStats({
                         packetsLost:        statsReport.packetsLost,
                         packetsCount:       statsReport.packetsReceived,
                         audioLevel:         when_defined(statsReport.audioInputLevel),
-                        rttMilliseconds:    streamType === 'audio_ouptut' || streamType === 'video_output' ? when_defined(statsReport.roundTripTime) : null,
+                        rttMilliseconds:    streamType === 'audio_output' || streamType === 'video_output' ? when_defined(statsReport.roundTripTime) : null,
                         jbMilliseconds:     streamType === 'audio_output' || streamType === 'video_output' ? when_defined(statsReport.jitter, 0) * 1000 : null
-                    };
+                    }, statsReport.type, streamType);
+                }
+            } else if(statsReport.type === 'VideoBwe') {
```

Comment:
The *if* is missing a space here and below.

Reply:
good catch